src/pykeen/models/predict.py

```diff
@@ -1,25 +1,164 @@
+- :func:`pykeen.models.predict.predict_target` can be used to score choices for a given prediction target, i.e.
+  calculate scores for head entities, relations, or tail entities given the other two.
+- :func:`pykeen.models.predict.predict` can be used to calculate scores for all possible triples.
+
+.. warning ::
+    Please note that not all models automatically have interpretable scores, and their calibration may be poor. Thus,
+    exercise caution when interpreting the results.
+
+
+Triple Scoring
+--------------
+
+When scoring triples with :func:`pykeen.models.predict.predict_triples`, we obtain a score for each of the given
+triples. As an example, we will calculate scores for all validation triples from the dataset we trained the model upon.
+
+>>> from pykeen.datasets import get_dataset
+>>> from pykeen.models.predict import predict_triples
+>>> dataset = get_dataset(dataset="nations")
+>>> pack = predict_triples(model=result.model, triples=dataset.validation)
+
+The variable :data:`pack` now contains a :class:`pykeen.models.predict.ScorePack`, which essentially is a pair of
+ID-based triples with their predicted scores. For interpretation, it can be helpful to add their corresponding labels,
+which the `"nations"` dataset offers, and convert them to a pandas dataframe:
+
+>>> df = pack.process(factory=result.training).df
+
+Since we now have a dataframe, we can utilize the full power of pandas for our subsequent analysis, e.g., showing the
+triples which received the highest score
+
+>>> df.nlargest(n=5, columns="score")
+
+or investigate whether certain entities generally receive larger scores
+
+>>> df.groupby(by=["head_id", "head_label"]).agg({"score": ["mean", "std", "count"]})
+
+
+Target Scoring
+--------------
+
+:func:`pykeen.models.predict.predict_target`'s primary usecase is link prediction or relation prediction.
+For instance, we could use our models to score all possible tail entities for the query `("uk", "conferences", ?)` via
+
+>>> from pykeen.datasets import get_dataset
+>>> from pykeen.models.predict import predict_target
+>>> dataset = get_dataset(dataset="nations")
+>>> pred = predict_target(
+...     model=result.model,
+...     head="uk",
+...     relation="conferences",
+...     triples_factory=result.training,
+... )
+
+Notice that the result stored into `pred` is a :class:`pykeen.models.predict.Predictions` object, which offers some
+post-processing options. For instance, we can remove all targets which are already know from the training set
+
+>>> pred_filtered = pred.filter_triples(dataset.training)
+
+or add additional columns to the dataframe proving the information whether the target is contained in another set,
+e.g., the validation or testing set.
+
+>>> pred_annotated = pred_filtered.add_membership_columns(validation=dataset.validation, testing=dataset.testing)
+
+The predictions object also exposes filtered / annotated dataframe through its `df` attribute
+
+>>> pred_annotated.df
```

Comment:
can you add notes about how these were done before to benefit people who need to update their code?

Reply:
https://github.com/pykeen/pykeen/pull/1090/commits/a5f791c935f14d46ab4413c744de026730df1092